### ISSUE TITLE
declaration-block-no-outline-without-alternative

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#1758]
Ref: https://github.com/CSSLint/csslint/wiki/Disallow-outline%3Anone

Does it adhere to our [criteria](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#let-everyone-know) for inclusion in core, or should it be a plugin?

**Comment by davidtheclark** [comment]
I don't think it does. There are an infinite number of accessible ways that you could style a focus state: the default outline is just one. The linter can't possibly know whether you've done a great job styling that focus state or not. And depending on your focus style, it might be a very good idea to turn off the default outline.

So I'd suggest that this be left to a plugin.

**Comment by jeddy3** [comment]
Agreed.

**Comment by ai** [comment]
I think the CSSLint rule cover others focus states. It raise on

```css
a:focus {
  outline: none;
}
```

But not raise on:

```css
a:focus {
  background: blue;
  outline: none;
}
```

**Comment by jeddy3** [comment]
> The linter can't possibly know whether you've done a great job styling that focus state or not

@ai It's this ambiguity that makes the rule a good candidate for a plugin.
